src/FeeCalculator.sol

```diff
@@ -342,17 +386,26 @@ contract FeeCalculator is IFeeCalculator, Ownable {
 
              Case exists only if asset pool domination is > 90% and amount is ~1e-18 of that asset in the pool
         */
-        if (feeSD < zero) {
+        if (feeSD < _zero) {
             return intoUint256(amountSD * dustAssetRedemptionRelativeFee);
         }
 
         return intoUint256(feeSD);
     }
 
-    /// @notice Returns the current fee setup.
-    /// @return recipients shares The fee recipients and their share of the total fee.
-    function getFeeSetup() external view returns (address[] memory recipients, uint256[] memory shares) {
-        recipients = _recipients;
-        shares = _shares;
+    function _calculateFee(
+        uint256 totalPoolSupply,
```

Comment:
Just a nit but I'd look into making the order of these integers consistent with how they are provided in `calculator`, otherwise it's a bit mind-boggling.